src/tools/unstable-book-gen/src/main.rs

```diff
@@ -112,14 +122,47 @@ fn copy_recursive(from: &Path, to: &Path) {
     }
 }
 
+fn collect_compiler_flags(rustc_path: impl AsRef<Path>) -> Features {
+    let mut rustc = Command::new(rustc_path.as_ref());
+    rustc.arg("-Zhelp");
```

Comment:
Rather than running -Zhelp and then trying to parse that, maybe we can parse the options! macro (https://github.com/rust-lang/rust/blob/040a98af70f0a7da03f3d5356531b28a2a7a77e4/compiler/rustc_session/src/options.rs#L2116)? I think that has all the -Z flags. That would also give you a file/line to link to. I'm not sure how valuable that is, but if it is, that seems like the way to get that information.

I'm OK with sticking with -Z/-C parsing since the syntax is both fairly simple and controlled by us, though.

We'd likely also want (perhaps in a separate PR) to enforce that all the -C flags in codegen options (also in the linked file) have documentation in the non-unstable book. That could be done in tidy.